src/common/Cargo.toml

```diff
@@ -84,6 +85,10 @@ criterion = "0.4"
 rand = "0.8"
 tempfile = "3"
 
+[build-dependencies]
+anyhow = "1"
+vergen = { version = "7.5", default-features = false, features = ["git"]}
```

Comment:
vergen will generate `cargo:rerun-if-changed` to avoid unnecessary recompiling.

Even though, when developing locally, doing `git commit` will trigger recompiling. To avoid that, we may abandon vergen and generate an envvar **in CI workflow script**, as described [here](https://stackoverflow.com/a/65972328/5739882).